terraform/main.tf

```diff
@@ -97,6 +99,8 @@ locals {
       osdisk_size_controller  = "500"
       num_builders_x86        = 1
       num_builders_aarch64    = 1
+      binary_cache_public_key = "ghaf-infra-dev:EdgcUJsErufZitluMOYmoJDMQE+HFyveI/D270Cr84I="
+      binary_cache_url        = "https://changeme"
```

Comment:
Minor detail at this point, but what do you think of setting this to 'http://localhost:8080' here, and removing the substituter setting from `hosts/azure/builder/configuration.nix` and `hosts/azure/jenkins-controller/configuration.nix`?

Reply:
For all machines that we define in Azure, and give (read-only) access to the bucket, I think they should definitely just use `http://localhost:8080`. There's no point in pulling that data from a neighboring machine. So for these VMs we should probably not set `extraNixSubstituter`, and just encode this in their `configuration.nix`. We define that `rclone-http` service right next to it, we know we have it there.

However, this per-environment URL should be the "public URL" (`https://` and what caddys' `SITE_ADDRESS` on the binary cache VM is pointed to). This can then be used to override `extraNixSubstituter` in places that are not in azure, or don't even have read-only access to the bucket (note `custom-nixos.nix` currently still is Azure-specific).

Reply:
Btw, I think some of the rclone-http setup should be living in a NixOS module, that might be enabled / configured from configuration.nix, but we shouldn't copypaste this too much into `configuration.nix`.